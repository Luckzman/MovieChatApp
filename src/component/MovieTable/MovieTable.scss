
.table-container{
  display: flex;
  justify-content: center;
  width: 100%;
  table {
    border-collapse: collapse;
    border-spacing: 0;
    max-width: 100%;
    border: 1px solid #ddd;
    border-radius: 5px;
    th, td {
      text-align: left;
      padding: 16px;
    }
    tbody{
      tr{
        &:hover{
          background-color: #e7e7e7;
        }
      }
    }
    tr:nth-child(even) {
      background-color: #f2f2f2
    }
    .title{
      cursor: pointer;
    }
  }
}