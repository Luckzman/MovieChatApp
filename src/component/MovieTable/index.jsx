import React, {Component} from 'react';
import CustomInput from '../CustomInput';
import FilterDropdown from '../FilterDropdown';
import Pagination from '../Pagination';
import './MovieTable.scss';

class MovieTable extends Component {

  state = {
    input: '',
    movie: this.props.movieData,
    pageOfItems: []
  }


  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
    this.filterMovie(e.target.value);
  }

  filterMovie = (input) => {
    console.log(this.state.movie.length, 'movielength')
    if(this.state.movie.length > 0) {
      this.setState(() => ({
        movie: this.props.movieData.filter((movie) => (movie.title.toLowerCase().includes(input.toLowerCase())))
      }))
    }
  }

  onChangePage = (pageOfItems) => {
    this.setState({ pageOfItems: pageOfItems });
  }

  filterByGenre = (value) => {
    const {movie} = this.state;
    this.setState(() => ({
      movie: movie.filter((movie) => (movie.genre.includes(value)))
    }))
  }
  
  render(){
    const {input, movie, pageOfItems} = this.state;
    const {triggerModal} = this.props;
    
    return(
      <>
        <div className="table-container">
          <table className="table">
            <thead className="">
              <tr className="">
                <th className="">Title</th>
                <th className="">Year</th>
                <th className="">Runtime</th>
                <th className="">Revenue</th>
                <th className="">Rating</th>
                <th className="">Genre</th>
              </tr>
            </thead>
            <tbody className="">
              <tr className="">
                <td className="">
                  <form onSubmit={this.handleSubmit}>
                    <CustomInput
                      name="input"
                      value={input}
                      onChange={this.handleChange}
                      placeholder="filter by title"
                    />
                  </form>
                </td>
                <td colSpan='4' className=""></td>
                <td className=""><FilterDropdown getDropdownValue={this.filterByGenre}/></td>
              </tr>
              {pageOfItems.map((movie, index) => {
                return(<tr key={`movie${index}`} className="table-body">
                  <td className="title" onClick={() => triggerModal(movie.title)}>{movie.title}</td>
                  <td className="">{movie.year}</td>
                  <td className="">{movie.runtime}</td>
                  <td className="">{movie.revenue}</td>
                  <td className="">{movie.rating}</td>
                  <td className="">{movie.genre.join(', ')}</td>
                </tr>)})
              }
            </tbody>
          </table>
        </div>
        <Pagination className="pagination" items={movie} onChangePage={this.onChangePage} />
      </>
    )
  }
}


export default MovieTable;
