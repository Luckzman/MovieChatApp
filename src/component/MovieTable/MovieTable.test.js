import React from 'react';
import { shallow } from 'enzyme';
import MovieTable from './';

describe('MovieTable', () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      triggerModal: jest.fn(),
      movieData: [{title: 'able'}]
    }
    wrapper = shallow(< MovieTable {...props} />);
  })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should be able to run onChangePage method when component is called', () => {
    expect(wrapper.instance().onChangePage).toBeInstanceOf(Function);
  });
  it('should be able to run filterByGenre method when component is called', () => {
    expect(wrapper.instance().filterByGenre).toBeInstanceOf(Function);
  });
  it('should be able to run filterMovie method when component is called', () => {
    expect(wrapper.instance().filterMovie).toBeInstanceOf(Function);
  });
  it('should be able to run handleChange method when component is called', () => {
    expect(wrapper.instance().handleChange).toBeInstanceOf(Function);
  });
});
