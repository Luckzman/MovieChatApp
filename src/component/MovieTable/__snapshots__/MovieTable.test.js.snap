// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTable should render correctly 1`] = `ShallowWrapper {}`;
